### PR TITLE
Implement GET_STATUS implementation result

### PR DESCRIPTION
**Pull request by pablogs9**
Signed-off-by: Pablo Garrido <pablogs9@gmail.com>

### CODE REVIEW

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/541/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/559/)

File: src/c/core/session/session.c [hunk 1]
```diff
@@ -646,6 +646,8 @@ bool uxr_acknack_pong(
             INFO_Payload info_payload;
 
             success &= uxr_deserialize_BaseObjectReply(buffer, &info_payload.base);
+            active_session = info_payload.base.result.implementation_status;
```
Comment (Acuadros95) [inline src/c/core/session/session.c:649]:
```suggestion
            active_session = info_payload.base.result.implementation_status;
            ret = active_session ? PONG_IN_SESSION_STATUS : PONG_NO_SESSION_STATUS;
```

File: src/c/core/session/session.c [hunk 2]
```diff
@@ -626,14 +626,14 @@ void uxr_flash_output_streams(
 //==================================================================
 //                             PRIVATE
 //==================================================================
-bool uxr_acknack_pong(
+pong_status_t uxr_acknack_pong(
         ucdrBuffer* buffer)
 {
     bool success = false;
     bool ret = false;
```
Comment (Acuadros95) [inline src/c/core/session/session.c:633]:
```suggestion
    pong_status_t ret = NO_PONG_STATUS;
```
Maybe we could simplify the return statement

File: src/c/core/session/session.c [hunk 3]
```diff
@@ -667,7 +669,7 @@ bool uxr_acknack_pong(
         }
     }
 
-    return ret;
+    return ret ? (active_session ? PONG_IN_SESSION_STATUS : PONG_NO_SESSION_STATUS) : NO_PONG_STATUS;
```
Comment (Acuadros95) [inline src/c/core/session/session.c:672]:
```suggestion
    return ret;
```

**Review by Acuadros95 — APPROVED** [review]

**Comment by pablogs9** [comment]
Tests here: https://github.com/eProsima/Micro-XRCE-DDS/pull/131

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/543/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/561/)

**Pull request merged**